### PR TITLE
Proposed white spacing changes

### PR DESCRIPTION
**Pull request by Neogavin**
...to make following guidelines easier without going excessively compulsive about how the files are formatted.

### CODE REVIEW

**Comment by wjramos** [comment]
I view `if`, `while`, `for` parentheses as characteristically different than function parentheses. I think the former reads more like the english language and do think a space makes more sense, while the latter is declaring something the function wholly depends on. I read no space as a function -- `while(_foo_)` feels kind of strange. However, for the sake of consistency I understand why it may be helpful to have it the same for both.

**Comment by peripateticus** [comment]
To be honest, I spend zero time concerned with white space formatting as I set up my IDE to format per the style guide on save or on a hotkey command.  (See https://github.com/beautify-web/js-beautify).   Assuming we are all using sublime, atom, or intellij, we should all be using the same .jsbeautifyrc file and be formatting automatically.  This, combined with the current breathing room not hurting readability, I'm likely not to vote for this update.

**Comment by Neogavin** [comment]
@peripateticus I'm really not sure if this should be considered a valid reason to -not- change the style guide.  Honestly, if the whole process can be completely automated, why do we even need a style guide in the first place?  We could just remove this and have it built into our FE dev build system to add in white space as needed.

But we'd still need to decide on what those white spaces are, so...

My argument is that, if it slows down actually writing the code, it should not be in place if its unnecessary.  In a perfect world we'd all just write completely minified Javascript, which would be the most efficient way, but since we don't, I think its good to specify how things should be written, regardless.  Otherwise this style guide could have any number of weird and nonsensical changes to it, and no one would bat an eye because we could just hit a hot key and conform to it.  I don't think that's right.

**Comment by Neogavin** [comment]
@wjramos This change is primarily for consistency's sake, and I'm not sure I agree on the "reading like the English language" portion.  If I write a function called `read`, does that suddenly have to be written as `read ( foo )` because its an actual word?  There's really no reason why those keywords should have a space after them besides preferences, and having to recall when to add a space and when not to just slows things down.

**Comment by peripateticus** [comment]
@Neogavin That's a good point..you're emphasizing efficiency of code writing. When we are writing app logic code, we shouldn't be *also* thinking about where we need to put a space or not; we should be adding space where it naturally feels right for us. I'm thinking the intentions of the liberal breathing room throughout the style guide were for readability but I know that this is a very subjective area. It may be just simply based on an assumption that leaning toward more whitespace is perceived as more readable than less, in general.  I know that's not the case on github for the function parameter breathing room that is in our style guide.  I suppose the question will come down to whether or not we favor 'writeability' over readability and also whether the additional space is perceived as more 'readable'.

**Comment by peripateticus** [comment]
@Neogavin To your point though, in the end, we need to agree on what's natural to us first, not let a style guide that is unnatural be imposed on us.

**Comment by peripateticus** [comment]
@wjramos I totally agree that we shouldn't be wasting *any* brain cycles thinking about space rules.  We need to come to an agreement on how we all want to *read* code or what is *readable* code.

**Comment by robatron** [comment]
@Neogavin , as discussed, take a look at ['space-before/after' ESLint rules](http://eslint.org/docs/rules/#stylistic-issues) to help define the guideline changes you're proposing as we'll soon be using ESLint to visualize style guide adherence.

**Comment by wjramos** [comment]
@Neogavin I simply meant I implicitly read `XXXX( stuff )` as a function

**Comment by wjramos** [comment]
More specifically `foo( bar )`, `bar` directly modifies `foo` and is essential for the operations it invokes.

In `if( bar )`, `bar` does not get eves passed to its contents in the context of its scope. It makes most sense to me to read it like 'If XXXX then XXXX'

**Comment by Neogavin** [comment]
@robatron Sorry, with all the redesign stuff last week, I never got a chance to put in those properties to clarify things.  If that should hold up the voting today I'm fine with it coming up next week instead.

**Comment by Neogavin** [comment]
@wjramos My only problem with reasoning like this is that you're using layout to define the difference between functions and a set of conditions.  I don't think a certain coding style should be used to differentiate between different types of functions/function calls like that.  What if someone comes along and claims that, instead of `if ( foo )`, its much easier for them to read `if     ( foo )` on their 4k monitor?  Or they decide they like two returns between function statements?  Or even...

Its all purely subjective, and I don't think things like that should be dictating our code.  I'm all for a balance of efficiency vs. readability, and I think this strikes a good balance.  But if its really a point of "This needs to be in here" then I can update this accordingly.

**Comment by Neogavin** [comment]
Trying to actually type formatting in the comments here on Git is such a huge hassle, lol.

**Comment by robatron** [comment]
@Neogavin , the intention of adding a description was to clarify the change proposal, so if everyone understands, it shouldn't hold up voting.

File: docs/javascript.md [hunk 1]
```diff
@@ -1452,7 +1452,7 @@ For more information see [Truth Equality and JavaScript](//javascriptweblog.word
 
     ...
 
-    $(this).on( 'listingUpdated', function ( e, data ) {
+    $(this).on( 'listingUpdated', function( e, data ) {
```
Comment (peripateticus) [inline docs/javascript.md]: Sidenote:  We're not following our breathing room rules in several places.

**Comment by peripateticus** [comment]
Closing old PRs.

**Pull request closed without merging**